src/util/support/plugins.c

```diff
@@ -592,9 +592,13 @@ krb5int_open_plugin_dirs (const char * const *dirnames,
                     }
                 }
 
-                if (krb5int_open_plugin (filepath, &handle, ep) == 0) {
-                    err = krb5int_plugin_file_handle_array_add (&h, &count, handle);
-                    if (!err) { handle = NULL; }  /* h takes ownership */
+                if (!err) {
+                    if (krb5int_open_plugin(filepath, &handle, ep) == 0) {
```

Comment:
I would prefer to combine these: ``if (!err && krb5int_open_plugin(...``

(This code should ideally be rewritten to use the currently favored ``goto cleanup`` style rather than an if-ladder.  But we should fix the bug minimally, and combining ``if (!err)`` conditionals with other checks appears to be normal in this file.)